### PR TITLE
fixing the race condition in the gui tests

### PR DESCRIPTION
**Pull request by emre2038**
## Description
`expandPanel` function in general.js used to create race condition because we were using a cypress function which works async in a loop.

this PR introduces a recursive function instead of a loop to fix the issue

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-445

### Updated
* `__tests__/gui/common_cy_functions/general.js` to implement a recursive function instead of a loop

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design match the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: __tests__/gui/common_cy_functions/general.js [hunk 1]
```diff
@@ -134,29 +135,50 @@ function expandPanel(panel) {
...
-    if (closedNoPanels) {
-        //nothing was expanded
-        cy.get(`.panel-toggles ${panelToBeExpanded.button}`).click({force: true});
-    }
+    const nonSelectedPanels = allPanels.filter((p) => {
+        return p.panel !== panel
+    });
+
+    recTogglePanel(panelToBeExpanded, nonSelectedPanels, 0).then(function (result) {
+        if (result) {
```
Comment (michaelglenister) [inline __tests__/gui/common_cy_functions/general.js:144]: Wouldn't `recTogglePanel` always return a truthy for `result`?

**Review by michaelglenister — COMMENTED** [review]
Just one question that needs to be looked at, otherwise it looks good, thanks

File: __tests__/gui/common_cy_functions/general.js (hunk 1 above)
Comment (emre2038) [reply __tests__/gui/common_cy_functions/general.js:144]: nope - if all the panels are already closed `recTogglePanel` returns false
https://github.com/OpenUpSA/wazimap-ng-ui/blob/506b442e77c3e11a25d939165b14bbded4a05ce4/__tests__/gui/common_cy_functions/general.js#L168

**Comment by michaelglenister** [comment]
That fixed most of the red CI but there's still this issue
`CypressError: Timed out retrying after 4050ms: `cy.click()` failed because this element is not visible:`

I think previously this was bypassed by using the `{force: true}` to force the click

Can you have a look at why this is happening

**Review by michaelglenister — COMMENTED** [review]
Nice work, ship it

**Pull request merged**